//
// Copyright 2018 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

// $title: Enclave Assertion Authority Configuration
// $overview: Configuration options for an enclave assertion authority.
// $location: https://asylo.dev/docs/reference/proto/identity/asylo.enclave_assertion_authority_config.v1.html
// $front_matter: redirect_from: /docs/reference/proto/identity/enclave_assertion_authority_config.html

// This package defines EnclaveAssertionAuthorityConfig, which is used to
// initialize an assertion authority (an assertion generator or an assertion
// verifier).
package asylo;

import "asylo/identity/identity.proto";

// Configuration used to initialize an enclave assertion authority (an assertion
// generator or an assertion verifier).
message EnclaveAssertionAuthorityConfig {
  // Describes the target enclave assertion authority.
  optional AssertionDescription description = 1;

  // A config to be used to initialize the authority. This field is an opaque
  // string that may take on any format. It could, for example, be a serialized
  // protobuf.
  optional bytes config = 2;
}
